apps/dashboard/src/pages/navbar-pages/staking/validator-details/ApyDetail.svelte

```diff
@@ -0,0 +1,56 @@
+<script lang="ts">
+  import { type UptimeValue } from '@api/utils/entities/component/validator'
+  import SkeletonLoader from '@components/_base/skeleton-loader/SkeletonLoader.svelte'
+  import { uptimeModule } from '@dashboard/lib/validators/uptime-module'
+  import { truncateNumber } from '@utils'
+  import { onMount } from 'svelte'
+  import type { TransformedValidator } from '../validator-list/ValidatorList.svelte'
+
+  export let validator: {
+    typed: {
+      type: 'String'
+      value: TransformedValidator
+    }
+  }
+
+  let selected: UptimeValue = '1month'
+  let isLoading: boolean
+  let validatorsUptimeData = uptimeModule.getDataForUptime(selected)
+
+  onMount(() => {
+    const subscription = uptimeModule.isLoading$.subscribe((data) => {
```

Comment:
I don't think you need to subscribe inside `onMount`

Reply:
If I do that I have `subscription` closured and easily accessibly for return function (onDestroy). Do you think it's any better to do not use `onMount` but use `onDestroy`?

Reply:
yes because there's no need to wait until after rendering (which is when `onMount` is called`). So it's better to just use `onDestroy` to clean up